server/routes/prisonerList/pagedPrisonerSearchSummary.ts

```diff
@@ -0,0 +1,219 @@
+import moment from 'moment'
+import type { PrisonerSearchSummary } from 'viewModels'
+
+/**
+ * A class encapsulating an array of [PrisonerSearchSummary] records, exposing them as a paged collection.
+ * The class exposes methods to sort, filter, change page and return the current page of records.
+ *
+ * All fields that represent indexes (`currentPageNumber`, `resultIndexFrom` and `resultIndexTo`) are 1 indexed to make
+ * it easier for view concerns to interact with this class.
+ *
+ * The internal state of this class is mutable. Of particular note is the internal array of [PrisonerSearchSummary] records
+ * which is mutated by the `filter` method. Calling the `filter` method removes elements from the internal array of
+ * [PrisonerSearchSummary] records, and this operation is not reversible.
+ */
+export default class PagedPrisonerSearchSummary {
+  private prisonerSearchSummaries: PrisonerSearchSummary[]
+
+  currentPageNumber: number
+
+  totalPages: number
+
+  pageSize: number
+
+  totalResults: number
```

Comment:
I'm not entirely happy with the level/access control of the encapsulation of these fields; but I guess this is more to do with the language, and that I cant help but compare it with what we'd do in java / kotlin.

By that I mean this (and the other properties here) are simple properties. They can be read publicly which is perfect:
```
if (pagedPrisonerSearchSummary.totalResults === 0) { 
  ....
```
but .... they can also be set publicly 😢 
```
pagedPrisonerSearchSummary.totalResults = 100
```

This annoys (the purist in) me!, and I did spend some time trying to get round it. The main problem is that javascript has no support for this, and typescript has limited support in the area of it's `readonly` modifier. Our fields here are not readonly - they are set internally at various stages as you apply filters, sorts, etc
The only way I could get it to work was to make the field private, then expose a getter method bot no setter method. It worked but it was kinda ugly, both in respect of the length of this class itself, plus the calling/consuming code (because it now has to call a getter method rather then simply reading the property)

In the end I decided to leave it as it is, accept that its not ideal (IMHO), and move on with life

Reply:
I think you may have been in danger of worrying unnecessarily! 😉

Reply:
You are 💯 right! 👍 😁 
I think the problem was that because it ended up being a class with behaviour, it kinda felt like I was doing it OO style in java or kotlin 😁 So you naturally start thinking like you would in java etc 😁 
But you are right, I'm worrying over nothing 😁